hswidget/dc6widget/state.go

```diff
@@ -5,12 +5,41 @@ import (
 	image2 "image"
 	"image/color"
 	"log"
+	"time"
 
 	"github.com/ianling/giu"
 
 	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2datautils"
 )
 
+const (
+	miliseconds     = 1000
+	defaultTickTime = 100
+)
+
+type animationPlayMode byte
+
+const (
+	playModeForward animationPlayMode = iota
+	playModeBackword
+	playModeLeftRight
```

Comment:
I think the name commonly given to this type of animation is `Yoyo`

Reply:
ee0bc54 (renamed to pingPong, because of string of this constant)